store/src/java/com/zimbra/cs/service/mail/Search.java

```diff
@@ -107,19 +107,17 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
             memberOfMap = ContactMemberOfMap.getMemberOfMap(mbox, octxt);
         }
 
-
-        ZimbraQueryResults results = mbox.index.search(zsc.getResponseProtocol(), octxt, params);
-        try {
-            // create the XML response Element
-            Element response = zsc.createElement(MailConstants.SEARCH_RESPONSE);
+        // create the XML response Element
+        Element response = zsc.createElement(MailConstants.SEARCH_RESPONSE);
+        try (ZimbraQueryResults results = mbox.index.search(zsc.getResponseProtocol(), octxt,
+            params)) {
             // must use results.getSortBy() because the results might have ignored our sortBy
             // request and used something else...
             response.addAttribute(MailConstants.A_SORTBY, results.getSortBy().toString());
             putHits(zsc, octxt, response, results, params, memberOfMap);
-            return response;
-        } finally {
-            Closeables.closeQuietly(results);
-        }
+        } catch (IOException e) {
```

Comment:
Why are you swallowine IOException here when weren't before?

Reply:
since that was the behavior with Closeables.closeQuietly(). It used to swallow any kind of Exception during close operation. When we are using try with resources, it makes it mandatory to catch the IOException if the resource close() method is declared to throw it.

Reply:
Ah.  Ok.  Thanks for the clarification.